pkg/mco/mc.go

```diff
@@ -0,0 +1,244 @@
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the MachineConfig is empty")
+
+		builder.errorMsg = "MachineConfig 'name' cannot be empty"
+	}
+
+	return &builder
+}
+
+// PullMachineConfig fetches existing machineconfig from cluster.
+func PullMachineConfig(apiClient *clients.Settings, name string) (*MCBuilder, error) {
+	glog.V(100).Infof("Pulling existing machineconfig name %s from cluster", name)
+
+	builder := MCBuilder{
+		apiClient: apiClient,
+		Definition: &mcv1.MachineConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the machineconfig is empty")
+
+		builder.errorMsg = "machineconfig 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("machineconfig object #{name} doesn't exist")
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Create builds machineconfig in the cluster and stores the created object in struct.
+func (builder *MCBuilder) Create() (*MCBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating MachineConfig %s", builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.MachineConfigs().Create(
+			context.TODO(), builder.Definition, metav1.CreateOptions{})
+	}
+
+	return builder, err
+}
+
+// Delete removes the machineconfig.
+func (builder *MCBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the MachineConfig object %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return fmt.Errorf("MachineConfig cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.MachineConfigs().Delete(
+		context.TODO(), builder.Object.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return fmt.Errorf("cannot delete MachineConfig :%w", err)
+	}
+
+	builder.Object = nil
+
+	return err
+}
+
+// Update renovates the existing machineconfig object with machineconfig definition in builder.
+func (builder *MCBuilder) Update() (*MCBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating machineconfig %s", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.MachineConfigs().Update(
+		context.TODO(), builder.Definition, metav1.UpdateOptions{})
+
+	return builder, err
+}
+
+// Exists checks whether the given machineconfig exists.
+func (builder *MCBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if the MachineConfig object %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.MachineConfigs().Get(
+		context.Background(), builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || k8serrors.IsNotFound(err)
```

Comment:
```suggestion
	return err == nil || !k8serrors.IsNotFound(err)
```

Reply:
Seems that I forgot to push it. 

Again, BIG Thanks !